### DISCUSSION TITLE
How to do 'shamefully-hoist' only in one package in a workspace?

### DISCUSSION THREAD

**Discussion by VanishMax** [pnpm/pnpm#4288] [Q&A] [answered]
I have a simple monorepo project with `apps` and `packages` folders. One of the apps uses a dependency that I need to hoist shamefully, but I don't want pnpm to do it for all other packages, which happens if I create per-project `.npmrc`. How can this logic be achieved?

**Comment by zkochan** [comment, marked as answer]
This is not possible to achieve because hoisting happens in the directory where the virtual store is located. And in a monorepo, there is a single virtual store at `<monorepo>/node_modules/.pnpm`. So you either have hoisting for all or for none.

But you can hoist specific packages using [`public-hoist-pattern`](https://pnpm.io/npmrc#public-hoist-pattern)

**Reply by VanishMax** [reply]
Thank you ❤️

**Reply by Mod5ied** [reply]
You could also use `pnpm i <your desired/missing package>` in place of a full front "shamefully-hoist".
💪

**Comment by leohxj** [comment]
can hoist specific packages using [public-hoist-pattern](https://pnpm.io/npmrc#public-hoist-pattern) in one package in a  workspace?
I try but failed, it seems like it only works with`.npmrc`in the root.

**Reply by zkochan** [reply]
no, as I said in the accepted answer, it is not possible. Only in the root of the workspace.
